app/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -31,20 +31,13 @@ export default function CodeSnippet({
     return <span className={styles.inlineCode}>{children}</span>;
   }
   if (children[children.length - 1] === "\n") children = children.slice(0, -1);
-  // remove the last \n from children
 
   if (highlightLineStart && !highlightLineEnd) {
     highlightLineEnd = highlightLineStart;
   }
 
   return (
-    <div
-      className={styles.codeSnippetContainer}
-      onCopy={(e) => {
-        e.preventDefault();
-        e.clipboardData.setData("text/plain", children);
-      }}
-    >
```

Comment:
Yes, it works! but have you seen the pasted code after this change? The code is not formatted correctly. It looks something like the below image
![image](https://github.com/user-attachments/assets/b50f36be-d010-40eb-85ea-a1488149e1a0)


We also need to keep the code formatted as it is in the codeblock. This formatting issue is primarily from the [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) library. You can look online and find out how other projects are using this library and find some solution for the issue.